src/core_plugins/i18n/server/__tests__/i18n.js

```diff
@@ -0,0 +1,158 @@
+import expect from 'expect.js';
+import i18n from '../i18n/i18n';
+import path from 'path';
+import Promise from 'bluebird';
+
+const PATH_SEPARATOR = path.sep;
+const DATA_PATH = __dirname + PATH_SEPARATOR + 'data';
+
+describe('Test registering translations for test_plugin_1', function () {
+  const pluginName = 'test_plugin_1';
+  const pluginTranslationPath = DATA_PATH + PATH_SEPARATOR + 'translations' + PATH_SEPARATOR + pluginName;
+
+  it('Register translations' , function (done) {
+    let result = true;
+    const translationFiles = [
+      pluginTranslationPath + PATH_SEPARATOR + 'de.json',
+      pluginTranslationPath + PATH_SEPARATOR + 'en.json'
+    ];
+
+    registerTranslations(translationFiles);
+    expect(result).to.be(true);
+    done();
+  });
+
+  it('EN translations are registered' , function (done) {
+    const locale = 'en';
+    const expectedTranslationJsonFile = DATA_PATH + PATH_SEPARATOR +
+      'reference' + PATH_SEPARATOR + pluginName + PATH_SEPARATOR + locale +
+      '.json';
+    const expectedTranslationJson = require(expectedTranslationJsonFile);
+    checkTranslations(locale, expectedTranslationJson, done);
+  });
+
+  it('DE translations are registered' , function (done) {
+    const locale = 'de';
+    const expectedTranslationJsonFile = DATA_PATH + PATH_SEPARATOR +
+      'reference' + PATH_SEPARATOR + pluginName + PATH_SEPARATOR + locale +
```

Comment:
Probably a throw back from early versions of the module. Makes sense to tidy it up. I have removed the reference files and now using inline strings.